internal/flagset/flagset.go

```diff
@@ -132,3 +105,48 @@ func (fs *Flagset) UnmarshalJSON(data []byte) error {
 
 	return nil
 }
+
+func LoadFromSourceFlags(data []byte) (*[]Flag, error) {
+	type SourceFlag struct {
+		Key string `json:"key"`
+		Type string `json:"type"`
+		Description string `json:"description"`
+		DefaultValue any `json:"defaultValue"`
+	}
+
+	var sourceFlags struct {
+		Flags []SourceFlag `json:"flags"`
+	}
+
+	if err := json.Unmarshal(data, &sourceFlags); err != nil {
+		return nil, err
+	}
+
+	var flags []Flag
+	for _, sf := range sourceFlags.Flags {
+		var flagType FlagType
+		switch sf.Type {
+		case "integer", "Integer":
+			flagType = IntType
+		case "float", "Float", "Number":
+			flagType = FloatType
+		case "boolean", "bool", "Boolean":
+			flagType = BoolType
+		case "string", "String":
+			flagType = StringType
+		case "object", "Object", "JSON":
+			flagType = ObjectType
+		default:
+			return nil, fmt.Errorf("unknown flag type: %s", sf.Type)
+		}
+
+		flags = append(flags, Flag{
+			Key:          sf.Key,
+			Type:         flagType,
+			Description:  sf.Description,
+			DefaultValue: sf.DefaultValue,
+		})
```

Comment:
similarily, this bit of logic feels redundant too. It looks like this func trying to do a lot of the same things as `UnmarshalJSON` above. Could we consolidate?